manager/integration/tests/test_basic.py

```diff
@@ -3658,3 +3660,86 @@ def test_volume_toomanysnapshots_condition(client, core_api, volume_name): # NOQ
+    3. Expand volume size to '2000000000/2G' and
+        check if size round up '2000683008/1908Mi'
+    4. Attach, write data, and detach
+    5. Expand volume size to '2Gi' and check if size is '2147483648'
+    6. Attach, write data, and detach
+    """
+
+    static_sc_name = "longhorn"
+    setting = client.by_id_setting(SETTING_DEFAULT_LONGHORN_STATIC_SC)
+    setting = client.update(setting, value=static_sc_name)
+    assert setting.value == static_sc_name
+
+    volume = create_and_check_volume(client, volume_name, 2, str(1 * Gi))
+    create_pv_for_volume(client, core_api, volume, volume_name)
+    create_pvc_for_volume(client, core_api, volume, volume_name)
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    volume = wait_for_volume_healthy(client, volume_name)
+    write_volume_random_data(volume)
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    volume.expand(size="2000000000")
+    wait_for_volume_expansion(client, volume_name)
+
+    for i in range(DEFAULT_POD_TIMEOUT):
+        claim = core_api.read_namespaced_persistent_volume_claim(
+            name=volume_name, namespace='default')
+        if claim.spec.resources.requests['storage'] == "2000683008" and \
+                claim.status.capacity['storage'] == "1908Mi":
+            break
+        time.sleep(DEFAULT_POD_INTERVAL)
+    assert claim.spec.resources.requests['storage'] == "2000683008"
+    assert claim.status.capacity['storage'] == "1908Mi"
+
+    volume = client.by_id_volume(volume_name)
+    assert volume.size == "2000683008"
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    volume = wait_for_volume_healthy(client, volume_name)
+    write_volume_random_data(volume)
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    volume.expand(size=str(2 * Gi))
+    wait_for_volume_expansion(client, volume_name)
+
+    for i in range(DEFAULT_POD_TIMEOUT):
+        claim = core_api.read_namespaced_persistent_volume_claim(
+            name=volume_name, namespace='default')
+        if claim.spec.resources.requests['storage'] == "2147483648" and \
+                claim.status.capacity['storage'] == "2Gi":
+            break
+        time.sleep(DEFAULT_POD_INTERVAL)
+    assert claim.spec.resources.requests['storage'] == "2147483648"
+    assert claim.status.capacity['storage'] == "2Gi"
+
+    volume = client.by_id_volume(volume_name)
+    assert volume.size == "2147483648"
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    volume = wait_for_volume_healthy(client, volume_name)
+    write_volume_random_data(volume)
```

Comment:
Need to verify existing data first.

Same below

Reply:
changed